### DISCUSSION TITLE
Application crash after Obfuscation in release build

### DISCUSSION THREAD

**Discussion by Connect-and-Exchange** [cefsharp/CefSharp#3937] [General]
After testing in debug mode the application and chromium browser are working fine. Once a release is build and the binary is obfuscated the app crashes the moment the chromium browser is opened in a WinForms Form.

the code is simple:

  ```
      UpgradeMessageForm upgradeForm;

        private void ShowReleaseNotes()
        {
            upgradeForm = new UpgradeMessageForm();
            upgradeForm.Show();
            upgradeForm.FormClosing += UpgradeForm_FormClosing;
        }

        private void UpgradeForm_FormClosing(object sender, FormClosingEventArgs e)
        {
            upgradeForm.FormClosing -= UpgradeForm_FormClosing;
            upgradeForm = null;
        }
```

code of the WinForms:

```
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace MegaCellMonitor
{
    public partial class UpgradeMessageForm : Form
    {
        public UpgradeMessageForm()
        {
            InitializeComponent();
            chromiumWebBrowser1.LoadUrl("https://www.megacellmonitor.com/ReleaseNotes_4_5_1_0");
        }
    }
}
```

It doesn't matter where i call it from, either through a user control like a button onClick or through a worker which calls the dispatcher and create a new task using invoke. 

eventviewer log:

Faulting application name: MegaCellMonitor.exe, version: 4.5.0.39, time stamp: 0x61c58939
Faulting module name: KERNELBASE.dll, version: 10.0.19041.1387, time stamp: 0x0b9a844a
Exception code: 0xe0434352
Fault offset: 0x0000000000034f69
Faulting process id: 0x3a64
Faulting application start time: 0x01d7f8a2fa797290
Faulting application path: C:\Projects\CellMonitor\bin\Release\MegaCellMonitor.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 7fae66de-001d-4131-a37e-e772ee3a0a5d
Faulting package full name: 
Faulting package-relative application ID: 

The application is deployed with a OneClick installer and can be found here: www.megacellmonitor.com

there is a beta and alpha release stream at https://www.megacellmonitor.com/alpha

**Comment by amaitland** [comment]
> Once a release is build and the binary is obfuscated the app crashes the moment the chromium browser is opened in a WinForms Form.

Does your `release build` work without obfuscation? If you are testing on a different machine make sure `Visual C++ 2019` or greater is installed.

> eventviewer log:
> 
> Faulting application name: MegaCellMonitor.exe, version: 4.5.0.39, time stamp: 0x61c58939 Faulting module name: KERNELBASE.dll, version: 10.0.19041.1387, time stamp: 0x0b9a844a Exception code: 0xe0434352

You'll need to contact the support team for the obfuscation tool you are using and ask how to get a meaningful exception. Make sure to mention that `CefSharp` uses `Visual C++ 2019`.
